engine/classes/ElggEntity.php

```diff
@@ -489,24 +489,23 @@ protected function setTempMetadata($name, $value, $multiple = false) {
 	 * @warning Calling this with no $name will clear all metadata on the entity.
 	 *
 	 * @param null|string $name The name of the metadata to remove.
-	 * @return bool|null
+	 *
+	 * @return bool
 	 * @since 1.8
 	 */
-	public function deleteMetadata($name = null) {
+	public function deleteMetadata($name = null): bool {
```

Comment:
should we make a deleteAllMetadata entity function to make deleting all metadata more explicit?

Reply:
maybe, but not in this PR